sdks/python/apache_beam/io/parquetio.py

```diff
@@ -565,22 +665,9 @@ def open(self, temp_path):
         use_compliant_nested_type=self._use_compliant_nested_type)
 
   def write_record(self, writer, value):
-    if len(self._buffer[0]) >= self._buffer_size:
-      self._flush_buffer()
-
-    if self._record_batches_byte_size >= self._row_group_buffer_size:
-      self._write_batches(writer)
-
-    # reorder the data in columnar format.
-    for i, n in enumerate(self._schema.names):
-      self._buffer[i].append(value[n])
+    writer.write_table(value)
```

Comment:
I have used this before where I infer the schema from the first batch that comes through. This simplified my model inference scripts and avoided loading tensorflow saved models to get output signature and converting them to pyarrow types.

It worked locally and on Dataflow. Thought I would throw it out there.